src/Validation/ValidatorAwareTrait.php

```diff
@@ -126,15 +126,6 @@ protected function createValidator(string $name): Validator
             $this->dispatchEvent($event, compact('validator', 'name'));
         }
 
-        if (!$validator instanceof Validator) {
-            throw new InvalidArgumentException(sprintf(
-                'The %s::%s() validation method must return an instance of %s.',
```

Comment:
This is another place where using assert() with a message would provide a better developer experience.